packages/xo-web/src/common/render-xo-item.js

```diff
@@ -40,6 +44,44 @@ const PoolObjectItem = propTypes({
   })
 )
 
+// VDI.
+@propTypes({
+  vdi: propTypes.object.isRequired,
+})
+@connectStore(() => {
+  const getSr = createGetObject((_, props) => props.vdi.$SR)
+  const getContainer = createGetObject(
+    createSelector(getSr, sr => sr.$container)
+  )
+
+  return (state, props) => ({
+    checkPermissions: getCheckPermissions,
+    container: getContainer(state, props),
+    sr: getSr(state, props),
+  })
+})
+class VdiItem extends Component {
+  _getCheckPermission = createSelector(
+    () => this.props.checkPermissions,
+    check => object => check(object.id, ['view', 'operate', 'administrate'])
+  )
+
+  render () {
+    const { vdi, sr, container } = this.props
+    return (
+      <span>
+        <Icon icon='disk' /> {vdi.name_label}
+        {this._getCheckPermission(sr) && (
```

Comment:
Why do you need to check the permissions here? Can't you just check if `sr` is `undefined`?